src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -45,6 +46,17 @@ public function getAlias()
     /**
      * {@inheritdoc}
      */
+    public function getConfiguration(array $config, ContainerBuilder $container)
+    {
+        // Will add the resource to the container
+        parent::getConfiguration($config, $container);
```

Comment:
http://symfony.com/doc/current/components/config/caching.html

The point is to execute the original method (https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/DependencyInjection/Extension/Extension.php#L91) which will not call the constructor itself but only add the resource.